File: ruby/3/LargestPrimeFactor.rb
#!/usr/bin/env ruby

$x = 600851475143
$i = 2
$factor = 0

while $x > 1
    if $x % $i == 0
        $x = $x / $i
        if $i > $factor
            $factor = $i
        end
    end
    $i += 1
end

puts $i

File: java/3/LargestPrimeFactor.java
class LargestPrimeFactor {
    public static void main(String[] args) {
        System.out.println(largestFactor(600851475143l));
    }

    public static int largestFactor(long x) {
        long y = x;
        int i = 2;
        int factor = 0;

        while(y > 1) {
            if (y % i == 0) {
                y = y / i;
                if (i > factor) {
                    factor = i;
                }
            }
            i++;
        }

        return factor;
    }
}
